array-reduce/main.js

```diff
@@ -0,0 +1,42 @@
+const numbers = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12];
+
+const account = [
+  { type: 'deposit', amount: 150 },
+  { type: 'deposit', amount: 20 },
+  { type: 'withdrawal', amount: 5 },
+  { type: 'deposit', amount: 100 },
+  { type: 'withdrawal', amount: 25 },
+  { type: 'withdrawal', amount: 60 }
+];
+
+const traits = [
+  { color: 'yellow' },
+  { type: 'electric' },
+  { name: 'pikachu' },
+  { level: 15 },
+  { trainer: 'ash' }
+];
+
+const sum = numbers.reduce((x, y) => x + y);
+
+console.log(sum);
+
+const product = numbers.reduce((x, y) => x * y);
+
+console.log(product);
+
+const balance = account.reduce((pre, cur) => {
+  if (cur.type === 'withdrawal') {
+    return pre - cur.amount;
+  }
+  return pre + cur.amount;
+}, 0);
+
+console.log(balance);
+
+const composite = traits.reduce((pre, cur) => {
+  return Object.assign(pre, cur);
+
+}, 0);
```

Comment:
Here the second argument to `reduce` should be an empty object `{}`. You want `pre` to start as an object.